### PR TITLE
remove analytics sample rate rule

### PR DESCRIPTION
**Pull request by richardstartin**

### CODE REVIEW

**Review by tylerbenson — APPROVED** [review]

**Review by tylerbenson — CHANGES_REQUESTED** [review]
Actually, I don't think we can do this since it's publicly documented to set as a tag: https://docs.datadoghq.com/tracing/legacy_app_analytics/?tab=java#custom-instrumentation

**Review by tylerbenson — APPROVED** [review]

File: dd-trace-core/src/main/java/datadog/trace/core/processor/TraceProcessor.java [hunk 1]
```diff
@@ -17,11 +16,7 @@
   final Rule[] DEFAULT_RULES =
       new Rule[] {
         // Rules are applied in order.
-        new DBStatementRule(),
-        new ResourceNameRule(),
-        new ErrorRule(),
-        new URLAsResourceNameRule(),
-        new AnalyticsSampleRateRule(),
+        new DBStatementRule(), new ResourceNameRule(), new ErrorRule(), new URLAsResourceNameRule()
```
Comment (lpriima) [inline dd-trace-core/src/main/java/datadog/trace/core/processor/TraceProcessor.java:19]: 👍  ?

**Review by lpriima — APPROVED** [review]

File: dd-trace-core/src/main/java/datadog/trace/core/processor/TraceProcessor.java (hunk 1 above)
Comment (richardstartin) [reply dd-trace-core/src/main/java/datadog/trace/core/processor/TraceProcessor.java:19]: spotless...

File: dd-trace-core/src/main/java/datadog/trace/core/CoreTracer.java [hunk 2]
```diff
@@ -673,6 +674,11 @@ public CoreSpanBuilder asChildOf(final AgentSpan agentSpan) {
     @Override
     public CoreSpanBuilder withTag(final String tag, final Object value) {
       switch (tag) {
+        case DDTags.ANALYTICS_SAMPLE_RATE:
+          if (value instanceof Number) {
```
Comment (bantonsson) [inline dd-trace-core/src/main/java/datadog/trace/core/CoreTracer.java:678]: So the old code tried to convert a `String` into a `Double`, but now we just drop it. Is that intended?

File: dd-trace-core/src/main/java/datadog/trace/core/DDSpanContext.java [hunk 3]
```diff
@@ -362,6 +363,11 @@ public void setTag(final String tag, final Object value) {
           this.spanType = (CharSequence) value;
         }
         break;
+      case ANALYTICS_SAMPLE_RATE:
+        if (value instanceof Number) {
```
Comment (bantonsson) [inline dd-trace-core/src/main/java/datadog/trace/core/DDSpanContext.java:367]: Here as well...

**Review by bantonsson — CHANGES_REQUESTED** [review]
Only one question about difference in behavior.

**Comment by richardstartin** [comment]
@bantonsson I thought that behaviour was silly so I removed it. I've put it back along with the tests.

**Review by bantonsson — APPROVED** [review]

**Pull request merged**